modules/apps/101-interchain-swap/keeper/msg_server.go

```diff
@@ -196,11 +195,16 @@ func (k Keeper) OnCreatePoolReceived(ctx sdk.Context, msg *types.MsgCreatePoolRe
 		return nil, types.ErrInvalidDenomPair
 	}
 
-	//TODO: Need to implement params module and market maker.
-	amm := types.NewInterchainMarketMaker(
-		&pool,
-		types.DefaultMaxFeeRate,
-	)
+	// Instantiate an interchain market maker with the default fee rate
+	amm, found := k.GetInterchainMarketMaker(ctx, pooId)
+	if !found {
+		amm = *types.NewInterchainMarketMaker(
+			&pool,
+			types.DefaultMaxFeeRate,
+		)
+		// save market maker
+		k.SetInterchainMarketMaker(ctx, amm)
+	}
```

Comment:
AMM should be stateless.  can be initiated by parameters.  not sure why store this object.